modules/clients/src/test/java/org/apache/ignite/internal/ServiceEventSubjectIdSelfTest.java

```diff
@@ -0,0 +1,301 @@
+    /** */
+    private static final String SVC_NAME = "simpleService";
+
+    /** */
+    private static final Collection<ServiceEvent> evts = new ArrayList<>();
+
+    /** */
+    private static UUID nodeId;
+
+    /** */
+    private static Ignite thickClient;
+
+    /** */
+    private static IgniteClient thinClient;
+
+    /** */
+    private static CountDownLatch latch;
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        IgniteConfiguration cfg = super.getConfiguration(igniteInstanceName);
+
+        cfg.setConnectorConfiguration(new ConnectorConfiguration());
+
+        cfg.setIncludeEventTypes(EventType.EVTS_SERVICE_EXECUTION);
+
+        cfg.setServiceConfiguration(
+                new ServiceConfiguration()
+                        .setName(SVC_NAME)
+                        .setService(new SimpleServiceImpl())
+                        .setTotalCount(1)
+        );
+
+        return cfg;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTestsStarted() throws Exception {
+        Ignite g = startGrid("srv_1");
+
+        g.events().localListen(new IgnitePredicate<Event>() {
+            @Override public boolean apply(Event evt) {
+                assert evt instanceof ServiceEvent;
+
+                evts.add((ServiceEvent)evt);
+
+                latch.countDown();
+
+                return true;
+            }
+        }, EVTS_SERVICE_EXECUTION);
+
+        thickClient = startClientGrid("cli_1");
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTestsStopped() throws Exception {
+
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTest() throws Exception {
+        evts.clear();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTest() throws Exception {
+
+    }
+
+    /**
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testSimpleServiceThinClient() throws Exception {
+        latch = new CountDownLatch(2);
+
+        ClientConfiguration cfg = new ClientConfiguration();
+
+        cfg.setAddresses("127.0.0.1:10801");
+
+        thinClient = Ignition.startClient(cfg);
+
+        nodeId = grid("srv_1").localNode()
+                .id();
+
+        ClientServices services = thinClient.services();
+
+        SimpleService simpleSvc = services.serviceProxy("simpleService", SimpleService.class, 1_000);
+
+        String simpleValue = simpleSvc.simpleMethod("simpleValue");
+
+        assertEquals("simpleValue", simpleValue);
+
+        assert latch.await(1_000, MILLISECONDS);
+
+        assertEquals(2, evts.size());
+
+        Iterator<ServiceEvent> it = evts.iterator();
+
+        assert it.hasNext();
+
+        ServiceEvent evt = it.next();
+
+        assert evt != null;
+
+        assertEquals(EVT_SERVICE_METHOD_EXECUTION_STARTED, evt.type());
+        assertEquals(nodeId, evt.subjectId());
```

Comment:
So, the test expects that the subject ID is the ID of the server node. Is it correct? Should it be the ID of the thin client? Just because the thin client is an initiating node. right?